core/base/dimensionReduction/DimensionReduction.cpp

```diff
@@ -70,6 +72,29 @@ int DimensionReduction::execute(
   Timer t;
 
   const int numberOfComponents = std::max(2, this->NumberOfComponents);
+  if(this->Method == METHOD::TOPOMAP) {
+    TopoMap topomap(
+      this->topomap_AngularSampleNb, topomap_CheckMST, topomap_Strategy);
+    topomap.setDebugLevel(this->debugLevel_);
+    topomap.setThreadNumber(this->threadNumber_);
+
+    double *ptrForCoordsTopomap = (double *)new double[2 * nRows];
```

Comment:
please do not, *ever*, use `new` or `delete`. my understanding is that this is considered as bad practice in modern C++.
in this example, use an std vector instead (`vector<double> coordsTopoMap(2*nRows)`) and the memory will be released from the heap at the end of the function (to pass the vector's buffer to topomap, simply do `coordsTopoMap.data()`)